modules/integration/src/test/scala/scala/cli/integration/ReplTests3StableDefinitions.scala

```diff
@@ -16,4 +20,31 @@ trait ReplTests3StableDefinitions { _: ReplTestDefinitions =>
       ammoniteTestScope(useMaxAmmoniteScalaVersion = false)
     }
   }
+
+  test("https://github.com/scala/scala3/issues/21229") {
+    TestInputs(
+      os.rel / "Pprint.scala" ->
+        """//> using dep "com.lihaoyi::pprint::0.9.0"
+          |package stuff
+          |import scala.quoted.*
+          |def foo = pprint(1)
+          |inline def bar = pprint(1)
+          |inline def baz = ${ bazImpl }
+          |def bazImpl(using Quotes) = '{ pprint(1) }
+          |""".stripMargin
+    ).fromRoot { root =>
+      val ammArgs = Seq("-c", "println(stuff.baz)")
+        .map {
+          if (Properties.isWin)
+            a => if (a.contains(" ")) "\"" + a.replace("\"", "\\\"") + "\"" else a
+          else
+            identity
+        }
+        .flatMap(arg => Seq("--ammonite-arg", arg))
+      // FIXME: test this on standard Scala 3 REPL, rather than just Ammonite
+      val res = os.proc(TestUtil.cli, "repl", ".", "--power", "--amm", ammArgs, extraOptions)
+        .call(cwd = root)
```

Comment:
Note: this is currently a test preventing accidentally breaking ammonite with this (which worked before the change), rather than an actual anti-regression test.
The Scala 3 REPL currently doesn't provide API that'd make this reasonably testable.

Reply:
Following this up needs:
- https://github.com/scala/scala3/issues/21242